data/core/contextmenu.lua

```diff
@@ -147,12 +132,27 @@ function ContextMenu:each_item()
   end)
 end
 
+function ContextMenu:on_mouse_pressed(button, x, y) 
+  if not self.visible then return false end
+  if x >= self.position.x and y >= self.position.y and x < self.position.x + self.items.width and y < self.position.y + self.height then
+    if button == 'left' then
```

Comment:
Nit: I guess the button check could be done above.

We definitely need an in-bounds check function.